openthread/src/scan.rs

```diff
@@ -135,17 +135,19 @@ impl OpenThread<'_> {
                 return Err(OtError::new(otError_OT_ERROR_BUSY));
             }
 
-            let f: &mut dyn FnMut(Option<&ScanResult>) = &mut f;
-
-            let scan_callback = &state.ot.scan_callback;
-            #[allow(clippy::missing_transmute_annotations)]
             {
-                *scan_callback.borrow_mut() = Some(unsafe { core::mem::transmute(f) });
-            }
+                let f: &mut dyn FnMut(Option<&ScanResult>) = &mut f;
 
-            let _guard = scopeguard::guard((), |_| {
-                *scan_callback.borrow_mut() = None;
-            });
+                let scan_callback = &mut state.ot.scan_callback;
+                #[allow(clippy::missing_transmute_annotations)]
```

Comment:
No big deal but this is the last remaining `clippy::missing_transmute_annotations` - would be nice to get rid of this